contracts/modules/capital/SwapOperator.sol

```diff
@@ -347,4 +361,162 @@ contract CowSwapOperator {
     uint feeInEther = oracle.getEthForAsset(asset, feeAmount);
     require(feeInEther <= maxFee, "SwapOp: Fee amount is higher than configured max fee");
   }
+
+
+  function swapETHForEnzymeVaultShare(uint amountIn, uint amountOutMin) external onlyController {
+
+    // Validate there's no current cow swap order going on
+    require(currentOrderUID.length == 0, "SwapOp: an order is already in place");
+
+    IPool pool = _pool();
+    IEnzymeV4Comptroller comptrollerProxy = IEnzymeV4Comptroller(IEnzymeV4Vault(enzymeV4VaultProxyAddress).getAccessor());
+    IERC20Detailed toToken = IERC20Detailed(enzymeV4VaultProxyAddress);
+
+
+    IPool.SwapDetails memory swapDetails = pool.getAssetSwapDetails(address(toToken));
+
+    require(!(swapDetails.minAmount == 0 && swapDetails.maxAmount == 0), "SwapOp: asset is not enabled");
+
+    {
+      // scope for swap frequency check
+      uint timeSinceLastTrade = block.timestamp - uint(swapDetails.lastSwapTime);
+      require(timeSinceLastTrade > MIN_TIME_BETWEEN_ORDERS, "SwapOp: too fast");
```

Comment:
Note i've reused the same MIN_TIME_BETWEEN_ORDERS used by cowswap.

In the v1 code this was 10 minutes.

This is 15. Seems find but comment if you think this is an issue.